### PR TITLE
Support for Windows infra-agent with log install

### PR DESCRIPTION
**Pull request by Julien4218**

### CODE REVIEW

**Review by jbeveland27 — DISMISSED** [review]
This is really great work @Julien4218! Left a few minor comments (and only the `name` in the recipe really needs addressing now). I think we should be able to merge this soon.

File: recipes/newrelic/infra-agent/windows.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+name: Infrastructure Agent Installer for Windows
+description: New Relic install recipe for the Infrastructure agent on Windows
+repository: https://github.com/newrelic/infrastructure-agent
+
+# See http://download.newrelic.com/infrastructure_agent/ for possible permutations
+installTargets:
+  - type: host
+    os: windows
```
[the remark this answers is not in the document]
Comment (jbeveland27) [reply recipes/newrelic/infra-agent/windows.yml:8]: @ctrombley If you get the chance, it'd be interesting to know what the CLI reports for `installTarget` when running on a Windows host. I'd suspect `platform` and `platformVersion` are empty.

File: test/deploy/assertions/recipe-is-valid/roles/onafterstart/tasks/main.yml [hunk 2]
```diff
@@ -24,8 +24,8 @@
     msg: "Using newrelic_api_url:{{ newrelic_api_url }}"
 
 - name: Get Hostname
-  shell: "hostname"
-  register: hostname_output
+  set_fact:
+    hostname_value: "{{ansible_hostname}}"
```
[the remark this answers is not in the document]
Comment (Julien4218) [reply test/deploy/assertions/recipe-is-valid/roles/onafterstart/tasks/main.yml:28]: Yeah, just found that out while working with windows and needing to support both linux and windows.

File: recipes/newrelic/infra-agent/windows.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
+name: Infrastructure Agent Installer for Windows
```
[the remark this answers is not in the document]
Comment (Julien4218) [reply recipes/newrelic/infra-agent/windows.yml:1]: @ctrombley and @zlesnr , what should the recipe name be for Infra on Windows?

[the remark this answers is not in the document]
Comment (zlesnr) [reply recipes/newrelic/infra-agent/windows.yml:1]: I'm not sure, but maybe the same thing its called for linux, and then let the other recipe attributes determine the platform.

File: recipes/newrelic/infra-agent/windows.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ctrombley) [reply recipes/newrelic/infra-agent/windows.yml:8]: https://github.com/shirou/gopsutil/blob/a1e77476b2bb7bbe6130fa1cbc39fe1635ff459c/v3/host/host_windows.go#L145-L187

[the remark this answers is not in the document]
Comment (Julien4218) [reply recipes/newrelic/infra-agent/windows.yml:8]: it looks like `os: windows` is provided by gopsutil.

**Review by devfreddy — APPROVED** [review]
LGTM 🎊

**Pull request merged**